libs/agno/agno/agent/agent.py

```diff
@@ -1081,13 +1106,32 @@ async def _arun(
         # 2. Generate a response from the Model (includes running function calls)
         model_response: ModelResponse = await self.model.aresponse(
             messages=run_messages.messages,
-            response_format=response_format,
             tools=self._tools_for_model,
             functions=self._functions_for_model,
             tool_choice=self.tool_choice,
             tool_call_limit=self.tool_call_limit,
+            **({"response_format": response_format} if self.parser_model is None else {}),
```

Comment:
why not just `response_format=response_format if self.parser_model is None else None`